### PR TITLE
it-tools: init at 2024.10.22-7ca5933

### LINKED ISSUE #348389: Package request: nixos/it-tools
**Project description**

Useful tools for developer and people working in IT.

**Metadata**

* homepage URL: https://it-tools.tech/
* source URL: https://github.com/CorentinTh/it-tools
* license: GPL-3.0
* platforms: unix

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by akotro**
Add it-tools, a website with useful tools for developers and people working in IT.

Fixes #348389 .

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by akotro** [comment]
Hello, this is my first time contributing to nixpkgs, so please let me know if there is anything I've missed, thanks!

File: pkgs/by-name/it/it-tools/package.nix [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    runHook preInstall
+
+    mkdir -p $out/lib
+    cp -R ./dist/* $out/lib/
+
+    runHook postInstall
+  '';
+
+  meta = {
+    description = "Collection of handy online tools for developers, with great UX.";
```
Comment (FliegendeWurst) [inline pkgs/by-name/it/it-tools/package.nix:47]: This description is not really useful.

Comment (akotro) [reply pkgs/by-name/it/it-tools/package.nix:47]: Agreed, maybe something like this?
"Static self-hostable website containing handy tools for developers, with great UX."

Comment (FliegendeWurst) [reply pkgs/by-name/it/it-tools/package.nix:47]: Sounds good. But make sure the description does not end with a period.

**Review by FliegendeWurst — COMMENTED** [review]
Can't review at the moment due to #353709

File: maintainers/maintainer-list.nix
Comment (Atemu) [inline maintainers/maintainer-list.nix:1]: Please split this into a separate commit.

**Review by Atemu — APPROVED** [review]
A few minor things and questions but the diff LGTM.

@74k1 could you verify that this works as expected?

File: pkgs/by-name/it/it-tools/package.nix [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    hash = "sha256-m1eXBE5rakcq8NGnPC9clAAvNJQrN5RuSQ94zfgGZxw=";
+  };
+
+  buildPhase = ''
+    runHook preBuild
+
+    pnpm build
+
+    runHook postBuild
+  '';
```
[the remark this answers is not in the document]
Comment (akotro) [reply pkgs/by-name/it/it-tools/package.nix:35]: From my testing, this does not seem to be the case, but I can obviously be corrected by someone more knowledgeable.

**Comment by 74k1** [comment]
Hi @Atemu 
Builds fine on my end (`nix build github:akotro/nixpkgs/it-tools#it-tools`) & has the expected output. I'm able to run & use it  with `http-server` in the `result/lib` path.
I would love to see a NixOS module / service coming along with it.
Great work so far! LGTM :heart:

**Comment by Atemu** [comment]
That's exactly what I wanted to know, thanks :)
